### PR TITLE
Task/enable modify pass4symkey and sslpassword

### PR DESCRIPTION
**Pull request by jrigassio-splunk**
Addresses #316 - The sslPassword option will be included in another issue / PR.

### CODE REVIEW

File: tests/small/test_environ.py [hunk 1]
```diff
@@ -77,33 +78,35 @@ def test_getSplunkPaths(default_yml, os_env, output):
...
-                ({"idxc": {"secret": "1234"}}, {}, {"label": None, "secret": "1234", "replication_factor": 1, "search_factor": 1}),
+                ({"idxc": {}}, {}, {"pass4SymmKey": None, "label": None, "secret": None, "replication_factor": 1, "search_factor": 1}),
+                ({"idxc": {"label": None}}, {}, {"pass4SymmKey": None, "label": None, "secret": None, "replication_factor": 1, "search_factor": 1}),
+                ({"idxc": {"label": "1234"}}, {}, {"pass4SymmKey": None, "label": "1234", "secret": None, "replication_factor": 1, "search_factor": 1}),
+                ({"idxc": {"secret": None}}, {}, {"pass4SymmKey": None, "label": None, "secret": None, "replication_factor": 1, "search_factor": 1}),
+                ({"idxc": {"secret": "1234"}}, {}, {"pass4SymmKey": "1234", "label": None, "secret": "1234", "replication_factor": 1, "search_factor": 1}),
+                ({"idxc": {"secret": None}}, {}, {"pass4SymmKey": None, "label": None, "secret": None, "replication_factor": 1, "search_factor": 1}),
+                ({"idxc": {"secret": "1234"}}, {}, {"pass4SymmKey": "1234", "label": None, "secret": "1234", "replication_factor": 1, "search_factor": 1}),
```
Comment (nwang92) [inline tests/small/test_environ.py]: This is probably my mistake, but lines 86-87 are identical to lines 88-89. Maybe we can change 88/89 from "secret" to "pass4SymmKey"?
```
{"idxc": {"pass4SymmKey": "1234"}}
```

Comment (jrigassio-splunk) [reply tests/small/test_environ.py]: Yep - the dangers of copy/paste. Will update

File: tests/small/test_environ.py [hunk 2]
```diff
@@ -117,28 +120,30 @@ def test_getIndexerClustering(default_yml, os_env, output):
...
-                ({"shc": {"secret": "1234"}}, {}, {"label": None, "secret": "1234", "replication_factor": 1}),
-                ({"shc": {"replication_factor": 0}}, {}, {"label": None, "secret": None, "replication_factor": 0}),
-                ({"shc": {"replication_factor": 1}}, {}, {"label": None, "secret": None, "replication_factor": 1}),
-                ({"shc": {"replication_factor": "2"}}, {}, {"label": None, "secret": None, "replication_factor": 2}),
+                ({"shc": {}}, {}, {"pass4SymmKey": None, "label": None, "secret": None, "replication_factor": 1}),
+                ({"shc": {"label": None}}, {}, {"pass4SymmKey": None, "label": None, "secret": None, "replication_factor": 1}),
+                ({"shc": {"label": "1234"}}, {}, {"pass4SymmKey": None, "label": "1234", "secret": None, "replication_factor": 1}),
+                ({"shc": {"secret": None}}, {}, {"pass4SymmKey": None, "label": None, "secret": None, "replication_factor": 1}),
+                ({"shc": {"secret": "1234"}}, {}, {"pass4SymmKey": "1234", "label": None, "secret": "1234", "replication_factor": 1}),
```
Comment (nwang92) [inline tests/small/test_environ.py]: Same comments re: test duplication here

Comment (nwang92) [reply tests/small/test_environ.py]: Nevermind, it's not a duplicate, but let's still add some pass4SymmKey tests.

File: roles/splunk_deployer/tasks/main.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@
     option: "{{ item.opt }}"
     value: "{{ item.val }}"
   with_items:
-    - { opt: 'pass4SymmKey', val: '{{ splunk.shc.secret }}' }
+    - { opt: 'pass4SymmKey', val: '{{ splunk.shc.pass4symkey }}' }
```
Comment (nwang92) [inline roles/splunk_deployer/tasks/main.yml]: Typo?

File: roles/splunk_common/tasks/main.yml [hunk 4]
```diff
@@ -40,6 +40,11 @@
   when:
     - java_version is defined and java_version
 
+- include_tasks: set_general_symmkey_password.yml
+  when:
+    - "'pass4SymmKey' in splunk"
```
Comment (nwang92) [inline roles/splunk_common/tasks/main.yml]: I'd also add an additional check to make sure splunk.pass4SymmKey is not null/empty so we don't needlessly call this play.

File: tests/small/test_environ.py (hunk 2 above)
Comment (jrigassio-splunk) [reply tests/small/test_environ.py]: doing just that

**Pull request merged**